packages/contracts/scripts/prebuild/smtchecker_prebuild.ts

```diff
@@ -0,0 +1,52 @@
+import fs from "fs";
+import path from "path";
+
+const foundry_file = path.join(__dirname, "../../foundry.toml");
+
+function getAllFiles(dirPath: string, arrayOfFiles: string[]): string[] {
+  const fullPath = path.resolve(dirPath);
+  const files = fs.readdirSync(fullPath);
+
+  for (const file of files) {
+    const filePath = path.join(fullPath, file);
+    const fileStat = fs.statSync(filePath);
+
+    if (fileStat.isDirectory()) {
+      getAllFiles(filePath, arrayOfFiles);
+    } else {
+      arrayOfFiles.push(filePath);
+    }
+  }
+
+  return arrayOfFiles;
+}
+
+function updateFoundryFile() {
+  const fileContent = fs.readFileSync(foundry_file, "utf8");
+  const contracts = getAllFiles("../../src", []).map((contract) =>
+    contract.replace("../../", "")
+  );
+
+  let contractsString = "";
+  for (const contract of contracts) {
+    const contractContent = fs.readFileSync(
+      path.join(__dirname, "../../", contract),
+      "utf8"
+    );
+    const namesMatched = contractContent.match(/contract\s+(\w+)\s+(is|{)/g);
```

Comment:
Perhaps it is better to use glob as @zgorizzo69 stated here https://github.com/ubiquity/ubiquity-dollar/pull/604#discussion_r1138332187 ? Searching in file content is brittle